### ISSUE TITLE
Sprint 6

### ISSUE THREAD

**Issue by dwinzg** [NyxHuang96/Fast-Repository-for-Analyzing-Unreliable-Discourse#83]
# COLX 581 - Under-represented languages

# Sprint 1

## Overall project goal
rubric={mechanics:1}

Last block, you created a small dataset - which is ideal for exploring methods in low-resource NLP.
This block, you will train and augment several machine learning tools to contextualize your corpus
with respect to data processing.

To keep things clear for grading, you should create a 581_Sprint_1 folder in your repo.
As always, people should be working on their own branches, and create a PR when something is ready for code review and merging.

## Teamwork contract
rubric={reasoning:1, writing:1}

Review and update your teamwork contract from last block.  
You should all read it again (it was only written 5 weeks ago, but do you remember what it said?),
and agree to changes that must be made in the contract.  Every team member should suggest
at least one change to the document.  Highlight these changes in the document, and resubmit to your team repo.  
Even if things worked well last block, this is a chance to explore new ways of working working around your many
commitments.  Be creative.

## Data splits
rubric={accuracy:1,quality:1}

Now that we are going to train machine learning, we need to ensure replicability and
be able to measure improvement.

Split your data into at least a training, development, and test set 
(you may also need other sets, but be careful - fragmenting data too much is costly...)

## Reasoned split (Optional)
rubric={raw:2}

It's often not enough to just split data into 80-10-10, and call it a day.
At a bare minimum, you should have no examples from your training set in test or dev
(or vice versa).  However, insidious information may leak across data sets: 

* Do you have the same speaker appear in multiple sets? 
* Do you have unrealistic class distributions in one or more set?
* Is one annotator disproportionately represented in a set?
* If your data is multilingual, are the languages fairly represented?
   * Does that fragment your data even more?
* Is your split replicable (and documented)?
   * If your hard-drive started sending out purple smoke, could you re-create the dataset?
 
You should do more than the bare minimum.  Make an effort to balance your data, and reasonable efforts to prevent leakage.  (If you're interested, [here's](https://aclanthology.org/2022.acl-short.96/) a paper about the dangers of data leakage...)

## Baselines
rubric={quality:2, efficiency:2, reasoning:2}

You are now pretty good at building machine-learning pipelines, 
and have a glut of tools at your disposal.  Use them!

We create a baseline using a simple model (remember MVPs?), training it on our train set,
and evaluating on our dev set 
(test sets don't get touched until after everything is implemented - it's too easy to 
overfit to our test set, if we're not careful...)

Create a data processing, training, and evaluation pipeline.  

You should pick two methods to train (one neural, one traditional?), and establish a
training pipeline.  You are free to use any of the tools we've talked about this year
(or others you know), except for LLMs / AI.  We'll get to those in later sprints...

Evaluate and report your results on your dev set.  It may help your report if you also
do a more in-depth analysis of the types of errors your model is making
(including qualitative analysis).

Document everything.

## Prompt completion
rubric={raw:4}

You should finish everything discussed above by 11:59pm Saturday, April 4. If you were not able to complete it on time you will lose 1 point from your timely completion score for each 12 hour period after the deadline. 
When you are done with a sprint (regardless of whether you are on time or late), you MUST open an issue in your repo that says you are done. Any modification of the relevant documents in your repo after you have opened this issue will likely result in losing all your prompt completion points.

A prompt completion grade of 0 (ie, a submission more than 2 days late or code modification after the deadline) will also result in a 50% penalty for the entire sprint.

**Comment by dwinzg** [comment]
- Sprint completed. 
- Release 2db2ee6.
